lib/web_ui/lib/ui_web/src/ui_web/navigation/platform_location.dart

```diff
@@ -81,13 +84,13 @@ class BrowserPlatformLocation implements PlatformLocation {
   DomHistory get _history => domWindow.history;
 
   @override
-  void addPopStateListener(DomEventListener fn) {
-    domWindow.addEventListener('popstate', fn);
+  void addPopStateListener(EventListener fn) {
+    domWindow.addEventListener('popstate', createDomEventListener(fn));
   }
 
   @override
-  void removePopStateListener(DomEventListener fn) {
-    domWindow.removeEventListener('popstate', fn);
+  void removePopStateListener(EventListener fn) {
+    domWindow.removeEventListener('popstate', createDomEventListener(fn));
```

Comment:
I just realized that this may not work because it creates a new function that's different from the one passed to `domWindow.addEventListener()`.

Does `Function.toJS` create a new JS function on each invocation? Or does it cache and return the same JS function? cc @joshualitt @eyebrowsoffire

Reply:
I ran into this too when I tried to migrate the plugins. `Function.toJS` creates a new function.

My 'fix' was to create a cache locally in this file to do the deduping. Another option is to change this API to take a `JSFunction`.

Reply:
This API used to take a `JSFunction` but I had to change it because we don't want to expose a JS type in the interface of the `PlatformLocation` class.

I think it's reasonable to have a cache inside the `BrowserPlatformLocation` class.